### ISSUE TITLE
POST payloads? non-200 responses?

### ISSUE THREAD

**Issue by mbilokonsky** [elderfo/microrouter-test-server#1]
Evaluating using this for a micro service I'm writing, but as far as I can tell a test fails on a non-200 response and it doesn't seem to want me to attach payloads to POST or PUT requests...? Are these cases just not supported yet, or am I doing it wrong?

**Comment by elderfo** [comment]
This uses [`request`](https://github.com/request/request#requestoptions-callback) (via [`request-promise`](https://github.com/request/request-promise)) under the covers and passes through any options specified. 

Here's an example Jest test that posts a payload (which is returned by the endpoint):
```js
const { post } = require('microrouter');
const { json, send } = require('micro');
const { createServer } = require('microrouter-test-server');

describe('post', () => {
  it('should return expected value', async () => {
    const server = await createServer([post('/post', async req => json(req))]);
    const payload = { abc: '123' };

    try {
      const result = await server.post('/post', {
        json: true,
        body: payload,
      });
      expect(result).toEqual(payload);
    } finally {
      server.close();
    }
  });
});
```

In regards to the non-200 responses, I found that the callbacks are not getting passed across to Request properly which could be part of the issue there. I am going to put more time into that and I will get back to you with a response.

**Comment by elderfo** [comment]
By default, any non-2xx response will throw an error (due to `request-promise`).

This can be turned off by setting the `simple` option to false: 
```js
const body = await server.post('/', { simple: false });
```
In this case `body` will contain the body contents of the response.

To get the whole response object, you can use the `resolveWithFullResponse` option:
```js
const response = await server.post('/', {
  simple: false,
  resolveWithFullResponse: true,
});
```

I hope that helps. I am going to be adding examples and test to show this behavior along with updating the readme.

**Comment by elderfo** [comment]
Re-opening as a reminder to update docs/tests

**Comment by elderfo** [comment]
Docs included in PR #2
